express-notes-api/write.js

```diff
@@ -0,0 +1,18 @@
+const fs = require('fs');
+
+function write(path, data) {
+  fs.writeFile(path, JSON.stringify(data, null, 2), { flag: 'r+' }, (err) => {
```

Comment:
I'm curious why you are using `r+` here? It just opens the file for both reading and writing, but since `writeFile` opens the file, writes, then closes the file, I don't think this is doing anything useful.

Reply:
From what I can remember, it was how I stopped `fs.writeFile()` from creating a new file if the path didn't exist. Or it was something involving receiving the 500 error along side the `console.error(err)`.